### PR TITLE
Input: pad_thread optimizations

### PR DESCRIPTION
**Pull request by Megamouse**
- Transforms pad_thread into a named_thread
- Adds a missing loop optimization that got lost during the last refactoring
- Increases sleep times while the thread is inactive and also adds sleep during emulation pause,
this reduces CPU usage while paused

### CODE REVIEW

File: rpcs3/Input/pad_thread.h [hunk 1]
```diff
@@ -35,9 +36,11 @@ class pad_thread
 	static std::shared_ptr<PadHandlerBase> GetHandler(pad_handler type);
 	static void InitPadConfig(cfg_pad& cfg, pad_handler type, std::shared_ptr<PadHandlerBase>& handler);
 
+	static auto constexpr thread_name = "Pad Thread"sv;
+	static constexpr u32 id_count = 1;
```
Comment (elad335) [inline rpcs3/Input/pad_thread.h:40]: What is this line for?

File: rpcs3/Input/pad_thread.cpp [hunk 2]
```diff
@@ -214,14 +214,15 @@ void pad_thread::SetIntercepted(bool intercepted)
...
-void pad_thread::ThreadFunc()
+void pad_thread::operator()()
 {
 	pad::g_active = true;
-	while (pad::g_active)
+
+	while (thread_ctrl::state() != thread_state::aborting)
 	{
-		if (!pad::g_enabled)
+		if (!pad::g_enabled || !Emu.IsRunning())
```
Comment (elad335) [inline rpcs3/Input/pad_thread.cpp:223]: IsPaused() should be used instead.

File: rpcs3/Input/pad_thread.cpp [hunk 3]
```diff
@@ -214,14 +214,15 @@ void pad_thread::SetIntercepted(bool intercepted)
...
 		{
-			std::this_thread::sleep_for(1ms);
+			std::this_thread::sleep_for(10ms);
```
Comment (elad335) [inline rpcs3/Input/pad_thread.cpp:225]: thread_ctrl::wait_for(10'000) because supports thread notifications.

File: rpcs3/Input/pad_thread.cpp [hunk 4]
```diff
@@ -40,15 +42,13 @@ pad_thread::pad_thread(void *_curthread, void *_curwindow, std::string_view titl
...
 
-	thread = std::make_shared<std::thread>(&pad_thread::ThreadFunc, this);
 	pad::g_current = this;
 }
 
 pad_thread::~pad_thread()
 {
 	pad::g_current = nullptr;
 	pad::g_active = false;
-	thread->join();
```
Comment (elad335) [inline rpcs3/Input/pad_thread.cpp:51]: Nothing stops you from joining the thread and it is needed, you need to assign thread_state::finished to it.

Comment (Megamouse) [reply rpcs3/Input/pad_thread.cpp:51]: why, we don't do it for any fxo thread

Comment (elad335) [reply rpcs3/Input/pad_thread.cpp:51]: Or actually just remove the following line `handlers.clear()` because of thread-safety issues.

Comment (Megamouse) [reply rpcs3/Input/pad_thread.cpp:51]: done

**Comment by Megamouse** [comment]
@elad335 you good with the changes?

**Comment by elad335** [comment]
g_active can be removed now I think.

**Comment by Megamouse** [comment]
I wanted to keep it for now to prevent regressions, but I can remove it I guess

**Comment by Megamouse** [comment]
done

**Review by elad335 — APPROVED** [review]

**Pull request merged**